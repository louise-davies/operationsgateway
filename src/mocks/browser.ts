import { setupWorker, rest, matchRequestUrl } from 'msw';
import { handlers } from './handlers';
import { browserHandlers } from './browserHandlers';

// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...handlers, ...browserHandlers);

// Make the `worker` and `rest` references available globally,
// so they can be accessed in both runtime and test suites.
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.msw = {
  worker,
  rest,
  matchRequestUrl,
};
