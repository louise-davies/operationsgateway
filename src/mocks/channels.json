{
  "_id": "123",
  "channels": {
    "CHANNEL_ABCDE": {
      "name": "Channel_ABCDE",
      "path": "/Channels/1/ABCDE",
      "type": "scalar"
    },
    "CHANNEL_BCDEF": {
      "name": "Channel_BCDEF",
      "path": "/Channels/1/BCDEF",
      "type": "image"
    },
    "CHANNEL_CDEFG": {
      "name": "Channel_CDEFG",
      "path": "/Channels/1/CDEFG",
      "type": "waveform",
      "x_units": "X",
      "y_units": "Y"
    },
    "CHANNEL_DEFGH": {
      "name": "Channel_DEFGH",
      "path": "/Channels/2/DEFGH",
      "type": "scalar",
      "notation": "scientific",
      "precision": 1
    },
    "CHANNEL_EFGHI": {
      "name": "Channel_EFGHI",
      "path": "/Channels/2/EFGHI",
      "type": "image"
    },
    "CHANNEL_FGHIJ": {
      "name": "Channel_FGHIJ",
      "path": "/Channels/2/FGHIJ",
      "type": "waveform"
    },
    "CHANNEL_GHIJK": {
      "name": "Channel_GHIJK",
      "path": "/Channels/3/GHIJK",
      "type": "scalar"
    }
  }
}
