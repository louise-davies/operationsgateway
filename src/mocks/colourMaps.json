{
  "Cyclic": ["hsv", "hsv_r", "twilight", "twilight_r"],
  "Diverging": ["BrBG", "BrBG_r", "bwr", "bwr_r"],
  "Miscellaneous": ["brg", "brg_r", "CMRmap", "CMRmap_r"],
  "Perceptually Uniform Sequential": [
    "cividis",
    "cividis_r",
    "inferno",
    "inferno_r"
  ],
  "Qualitative": ["Accent", "Accent_r", "Dark2", "Dark2_r"],
  "Sequential": ["Blues", "Blues_r", "BuGn", "BuGn_r"],
  "Sequential (2)": ["afmhot", "autumn", "autumn_r"]
}
