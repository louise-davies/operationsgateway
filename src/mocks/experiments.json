[
  {
    "_id": "19210001-1",
    "end_date": "2022-01-03T12:00:00",
    "experiment_id": "19210001",
    "part": 1,
    "start_date": "2021-12-31T13:00:00"
  },
  {
    "_id": "19210001-2",
    "end_date": "2022-01-06T12:00:00",
    "experiment_id": "19210001",
    "part": 2,
    "start_date": "2022-01-03T13:00:00"
  },
  {
    "_id": "19210012-1",
    "end_date": "2022-01-09T12:00:00",
    "experiment_id": "19210012",
    "part": 1,
    "start_date": "2022-01-06T13:00:00"
  },
  {
    "_id": "19210012-2",
    "end_date": "2022-01-12T12:00:00",
    "experiment_id": "19210012",
    "part": 2,
    "start_date": "2022-01-09T13:00:00"
  },
  {
    "_id": "22110007-1",
    "end_date": "2022-01-15T12:00:00",
    "experiment_id": "22110007",
    "part": 1,
    "start_date": "2022-01-12T13:00:00"
  },
  {
    "_id": "22110007-2",
    "end_date": "2022-01-18T12:00:00",
    "experiment_id": "22110007",
    "part": 2,
    "start_date": "2022-01-15T13:00:00"
  }
]
