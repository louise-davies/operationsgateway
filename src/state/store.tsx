import { configureStore } from '@reduxjs/toolkit';
import OperationsGatewayMiddleware, {
  listenToMessages,
} from './middleware/operationsgateway.middleware';
import configReducer from './slices/configSlice';

export const store = configureStore({
  reducer: {
    config: configReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(OperationsGatewayMiddleware),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

listenToMessages(store.dispatch);
