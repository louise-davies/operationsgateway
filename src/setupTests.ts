// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import crypto from 'crypto';
// TODO JOEL: Is this still needed? - Also in vite config
// need to mock <canvas> for plotting
import 'vitest-canvas-mock';
// TODO JOEL: Replace with vitest-fail-on-console
import failOnConsole from 'jest-fail-on-console';
import { TextEncoder } from 'util';
import { server } from './mocks/server';

// TODO JOEL: Is this still needed? - same for failOnConsole below
global.TextEncoder = TextEncoder;

failOnConsole();

vi.setConfig({ testTimeout: 15000 });

// Establish API mocking before all tests.
beforeAll(() => server.listen());

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());

// Clean up after the tests are finished.
afterAll(() => server.close());

// TODO JOEL: Check if a all of the below is still needed

if (typeof window.URL.createObjectURL === 'undefined') {
  // required as work-around for enzyme/jest environment not implementing window.URL.createObjectURL method
  Object.defineProperty(window.URL, 'createObjectURL', {
    value: () => 'testObjectUrl',
  });
}

if (typeof window.URL.revokeObjectURL === 'undefined') {
  // required as work-around for enzyme/jest environment not implementing window.URL.createObjectURL method
  Object.defineProperty(window.URL, 'revokeObjectURL', {
    value: () => {},
  });
}

// this is needed because of https://github.com/facebook/jest/issues/8987
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
let mockActualReact;
vi.doMock('react', async () => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  if (!mockActualReact) {
    mockActualReact = await vi.importActual('react');
  }
  return mockActualReact;
});

if (typeof window.URL.createObjectURL === 'undefined') {
  // required as work-around for enzyme/jest environment not implementing window.URL.createObjectURL method
  Object.defineProperty(window.URL, 'createObjectURL', {
    value: () => 'testObjectUrl',
  });
}

if (typeof window.URL.revokeObjectURL === 'undefined') {
  // required as work-around for enzyme/jest environment not implementing window.URL.revokeObjectURL method
  Object.defineProperty(window.URL, 'revokeObjectURL', {
    value: () => {
      // no-op
    },
  });
}

// jest doesn't implement web crypto so set up nodejs crypto as a default
Object.defineProperty(global, 'crypto', {
  value: Object.setPrototypeOf({ subtle: crypto.subtle }, crypto),
});
