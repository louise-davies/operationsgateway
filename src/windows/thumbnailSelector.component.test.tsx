import React from 'react';
import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/react';
import { getInitialState, renderComponentWithProviders } from '../setupTests';
import ThumbnailSelector from './thumbnailSelector.component';
import { http, HttpResponse } from 'msw';
import { server } from '../mocks/server';
import { RootState } from '../state/store';

describe('Thumbnail selector component', () => {
  const changeRecordId = jest.fn();
  let channelName;
  let recordId;

  beforeEach(() => {
    channelName = 'CHANNEL_BCDEF';
    recordId = '4';
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const createView = (preloadedState?: Partial<RootState>) => {
    return renderComponentWithProviders(
      <ThumbnailSelector
        channelName={channelName}
        recordId={recordId}
        changeRecordId={changeRecordId}
      />,
      { preloadedState }
    );
  };

  it('renders list of thumbnails and pagination controls', async () => {
    const view = createView();

    await screen.findAllByRole('img');

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const loadingHandler = () => {
      // taken from https://github.com/mswjs/msw/issues/778 - a way of mocking pending promises without breaking jest
      return new Promise(() => undefined);
    };
    server.use(
      http.get('/records', loadingHandler),
      http.get('/records/count', loadingHandler),
      http.get('/channels', loadingHandler)
    );

    const view = createView();
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('can change results per page and current page', async () => {
    const user = userEvent.setup();

    createView();
    await screen.findAllByRole('img');

    expect(screen.getByText('1–18 of 18')).toBeInTheDocument();

    await user.click(screen.getByRole('combobox', { name: 'Page size' }));
    await user.click(screen.getByRole('option', { name: '10' }));

    expect(screen.getByText('1–10 of 18')).toBeInTheDocument();

    await user.click(screen.getByRole('button', { name: 'Go to next page' }));

    expect(screen.getByText('11–18 of 18')).toBeInTheDocument();
  });

  it('clicking on a thumbnail calls changeRecordId', async () => {
    channelName = 'CHANNEL_CDEFG';
    recordId = '7';
    const user = userEvent.setup();

    createView();
    const thumbnails = await screen.findAllByRole('img');

    await user.click(thumbnails[0]);

    expect(changeRecordId).toHaveBeenCalledWith('7');
  });

  it('allows selection of 100 results when max shots is not 50', async () => {
    const user = userEvent.setup();

    createView({
      search: {
        searchParams: {
          ...getInitialState().search.searchParams,
          maxShots: 1000,
        },
      },
    });
    await screen.findAllByRole('img');

    await user.click(screen.getByRole('combobox', { name: 'Page size' }));
    expect(screen.getByRole('option', { name: '100' })).toBeInTheDocument();
  });

  it('displays max shots as the max pagination when record count is more than it', async () => {
    server.use(
      http.get('/records/count', () => {
        return HttpResponse.json(100, { status: 200 });
      })
    );

    createView();
    await screen.findAllByRole('img');

    expect(screen.getByText('1–25 of 50')).toBeInTheDocument();
  });
});
