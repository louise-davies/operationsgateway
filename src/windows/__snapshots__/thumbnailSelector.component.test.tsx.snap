// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail selector component renders correctly when loading 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-1r907rv-MuiGrid-root"
  >
    <ul
      class="MuiList-root css-zg5ruu-MuiList-root"
    />
    <div
      class="MuiTablePagination-root css-u4fheo-MuiTablePagination-root"
      padding="none"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r7:"
        >
          Page size
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-6ciqo7-MuiInputBase-root-MuiTablePagination-select"
        >
          <div
            aria-controls=":r8:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r7: :r6:"
            class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id=":r6:"
            role="combobox"
            tabindex="0"
          >
            25
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="25"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“25 of more than 25
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeSmall css-bmuri4-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-bmuri4-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Thumbnail selector component renders list of thumbnails and pagination controls 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-1r907rv-MuiGrid-root"
  >
    <ul
      class="MuiList-root css-zg5ruu-MuiList-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <img
          alt="CHANNEL_BCDEF image for timestamp 2022-01-04T00:00:00"
          aria-label=""
          class=""
          data-mui-internal-clone-element="true"
          src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIABkAGQMBIgACEQEDEQH/xAAbAAABBAMAAAAAAAAAAAAAAAAFAAYICQECB//EACYQAAICAgICAQQDAQAAAAAAAAECAwQFBhESABMHISIxQRQVFkL/xAAUAQEAAAAAAAAAAAAAAAAAAAAA/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8ArB1DXTt+24bBLfpYpsnchpi9kpvTWr+xwvslf/lF55J/QB8fGX+GasFvP2MbscN3XtetNUy+SlgCPXYSMkZSNXYTLKVIjKN9SD39a/cR2iYbGZHDM9ajjc3szW2iGOy981Ilg6p0eIB4vbIzl16+wkAD7G7crYLgdP0nIfEuxZLNQXa23x2n9mLpasktC4/EX9lJJIa7FllYS9GEqnotbpzLyGCtXb8BFrWbNOvae7XetWtxTyQiJ2SaCOZeyBmCkCQAgMRyD9fA3kg/mqvrNbHbjHl9bxeK2d7dSXXr2Nyzu7UQAEhlpB5BFzXeJgz+vj1qqqeWIj54BrSs3DrO54HMWI3lr4/IV7ciRcdmWORWIHP054H78f2U+QdO1nHf5vUKuYyeq5anWfYTm/VXuWraDk+hoy6xRxP2aPkMSWIk7gADlR/J8x4B/dM1Qy9rHR47+TJWo0o6gs3EVJp+pYhmVWYDqGCAdj9sa/j8BvebeLwP/9k="
          style="margin: auto; border: 1px solid #000000;"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <img
          alt="CHANNEL_BCDEF image for timestamp 2022-01-05T00:00:00"
          aria-label=""
          class=""
          data-mui-internal-clone-element="true"
          src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIABkAGQMBIgACEQEDEQH/xAAbAAABBAMAAAAAAAAAAAAAAAAFAAYICQECB//EACYQAAICAgICAQQDAQAAAAAAAAECAwQFBhESABMHISIxQRQVFkL/xAAUAQEAAAAAAAAAAAAAAAAAAAAA/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8ArB1DXTt+24bBLfpYpsnchpi9kpvTWr+xwvslf/lF55J/QB8fGX+GasFvP2MbscN3XtetNUy+SlgCPXYSMkZSNXYTLKVIjKN9SD39a/cR2iYbGZHDM9ajjc3szW2iGOy981Ilg6p0eIB4vbIzl16+wkAD7G7crYLgdP0nIfEuxZLNQXa23x2n9mLpasktC4/EX9lJJIa7FllYS9GEqnotbpzLyGCtXb8BFrWbNOvae7XetWtxTyQiJ2SaCOZeyBmCkCQAgMRyD9fA3kg/mqvrNbHbjHl9bxeK2d7dSXXr2Nyzu7UQAEhlpB5BFzXeJgz+vj1qqqeWIj54BrSs3DrO54HMWI3lr4/IV7ciRcdmWORWIHP054H78f2U+QdO1nHf5vUKuYyeq5anWfYTm/VXuWraDk+hoy6xRxP2aPkMSWIk7gADlR/J8x4B/dM1Qy9rHR47+TJWo0o6gs3EVJp+pYhmVWYDqGCAdj9sa/j8BvebeLwP/9k="
          style="margin: auto; border: 1px solid #000000;"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <img
          alt="CHANNEL_BCDEF image for timestamp 2022-01-06T00:00:00"
          aria-label=""
          class=""
          data-mui-internal-clone-element="true"
          src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAAwICAwICAwMDAwQDAwQFCAUFBAQFCgcHBggMCgwMCwoLCw0OEhANDhEOCwsQFhARExQVFRUMDxcYFhQYEhQVFP/bAEMBAwQEBQQFCQUFCRQNCw0UFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFP/AABEIABkAGQMBIgACEQEDEQH/xAAbAAABBAMAAAAAAAAAAAAAAAAFAAYICQECB//EACYQAAICAgICAQQDAQAAAAAAAAECAwQFBhESABMHISIxQRQVFkL/xAAUAQEAAAAAAAAAAAAAAAAAAAAA/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8ArB1DXTt+24bBLfpYpsnchpi9kpvTWr+xwvslf/lF55J/QB8fGX+GasFvP2MbscN3XtetNUy+SlgCPXYSMkZSNXYTLKVIjKN9SD39a/cR2iYbGZHDM9ajjc3szW2iGOy981Ilg6p0eIB4vbIzl16+wkAD7G7crYLgdP0nIfEuxZLNQXa23x2n9mLpasktC4/EX9lJJIa7FllYS9GEqnotbpzLyGCtXb8BFrWbNOvae7XetWtxTyQiJ2SaCOZeyBmCkCQAgMRyD9fA3kg/mqvrNbHbjHl9bxeK2d7dSXXr2Nyzu7UQAEhlpB5BFzXeJgz+vj1qqqeWIj54BrSs3DrO54HMWI3lr4/IV7ciRcdmWORWIHP054H78f2U+QdO1nHf5vUKuYyeq5anWfYTm/VXuWraDk+hoy6xRxP2aPkMSWIk7gADlR/J8x4B/dM1Qy9rHR47+TJWo0o6gs3EVJp+pYhmVWYDqGCAdj9sa/j8BvebeLwP/9k="
          style="margin: auto; border: 1px solid #000000;"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </ul>
    <div
      class="MuiTablePagination-root css-u4fheo-MuiTablePagination-root"
      padding="none"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r1:"
        >
          Page size
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-6ciqo7-MuiInputBase-root-MuiTablePagination-select"
        >
          <div
            aria-controls=":r2:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r1: :r0:"
            class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id=":r0:"
            role="combobox"
            tabindex="0"
          >
            25
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="25"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“18 of 18
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeSmall css-bmuri4-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeSmall css-bmuri4-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
